library/lms.c

```diff
@@ -63,8 +63,8 @@
                                      MBEDTLS_LMOTS_I_KEY_ID_LEN)
 
 
-/* Currently only support H=10 */
-#define H_TREE_HEIGHT_MAX                  10
+/* Currently only support H=10 and H=20 */
+#define H_TREE_HEIGHT_MAX                  20
 #define MERKLE_TREE_NODE_AM_MAX            (1u << (H_TREE_HEIGHT_MAX + 1u))
```

Comment:
This causes issues, putting 64M on the stack is non-trivial. Probably needs a malloc?

Reply:
Agreed, several things will have to switch using to malloc. When you do this you'll likely be able to remove some of the `_MAX` macros, since those are only there to support stack allocation.

You're likely to run into more issues with H=20 as well (I think you typoed it to H=10 on the original email so I didn't mention them there). One particular one is that the private key operations aren't efficient at large tree sizes as they always build the whole tree. There are various optimizations that other implementations make to support this, but I excluded them for simplicity's sake. Likely the largest problem we're going to have is that the tests for this might take an unacceptably long time.

Reply:
We've been thinking of splitting out some intensive testing that doesn't really depend on the configuration, and running those tests only once rather than in all configurations. Things falling into the gap between the unit tests that we want to run on a casual `make `test`, and fuzzing. But so far we haven't had a compelling use case.

One possibility would be to put this testing into a companion program, with a demo script (on the model of `programs/psa/key_ladder_demo.sh`) that runs only on one CI configuration.

Reply:
I have some local code to support M24_H10 which if we add would be maybe a better test to run to check the M24 differences since it doesn't take hours to run.

I can tack this onto this pull request or create a new one. WDYT?